dpnp/dparray.pyx

```diff
@@ -45,19 +45,98 @@ import numpy
 cimport numpy
 
 cimport dpnp.dpnp_utils as utils
+cimport dpctl as c_dpctl
+cimport dpctl.memory as c_dpctl_mem
+import dpctl
+
+cdef class _ArrayInterfaceData:
+    cdef int nd
+    cdef char * pointer
+    cdef Py_ssize_t itemsize
+    cdef int writeable
+    cdef tuple shape
+    cdef tuple strides
+    cdef object dtype
+    cdef c_dpctl.SyclQueue queue
+
+    @staticmethod
+    cdef _ArrayInterfaceData parse_sycl_usm_array_interface(object memory):
+        cdef dict iface
+        cdef _ArrayInterfaceData res_iface
+        cdef sycl_ctx
+        cdef char * iface_pointer
+        
+        if not hasattr(memory, "__sycl_usm_array_interface__"):
+            raise TypeError("Memory object does not expose __sycl_usm_array_interface__")
```

Comment:
use predefined functions from `utility` to raise exceptions

Reply:
The only free-form exceptions there is for run-time error. These are all `TypeError`s. 
The `checker_throw_type_error` is not sufficiently flexible in wording. 

Is the intent of suggestion to conform to a uniform style of exception reporting?